### PR TITLE
Update repo structure and all examples

### PR DESCRIPTION
**Pull request by brikis98**
This PR is a major refactor of this repo:

1. Switch to the full `examples` / `modules` / `test` folder structure we recommend.
2. Add simple, example modules for S3, Lambda, and ECS Fargate.
3. Add examples and automated tests for all modules.
4. Add CircleCi config to run tests.

### CODE REVIEW

**Comment by brikis98** [comment]
> #### ⚠️ GitGuardian has uncovered 1 secret following the scan of your pull request.
> Please consider investigating the findings and remediating the incidents. Failure to do so may lead to compromising the associated services or software components.
> 
> 🔎 Detected hardcoded secret in your pull request
> 🛠 Guidelines to remediate hardcoded secrets
> 🦉 [GitGuardian](https://dashboard.gitguardian.com/auth/login/?utm_medium=checkruns&utm_source=github&utm_campaign=cr1) detects secrets in your source code to help developers and security teams secure the modern development process. You are seeing this because you or someone else with access to this repository has authorized GitGuardian to scan your pull request.Our GitHub checks need improvements? [Share your feedbacks](https://form.typeform.com/to/KmeAPTMk)!

This is a false positive. Safe to ignore.

File: examples/ecs-fargate-service/.terraform.tfstate.lock.info [hunk 1]
```diff
@@ -0,0 +1 @@
+{"ID":"8389cea4-8f76-b412-193a-7957e928771c","Operation":"OperationTypeApply","Info":"","Who":"brikis98@Yevgeniys-MacBook-Air.local","Version":"1.6.2","Created":"2024-03-07T18:41:05.861475Z","Path":"terraform.tfstate"}
```
Comment (Resonance1584) [inline examples/ecs-fargate-service/.terraform.tfstate.lock.info:1]: Is this file included intentionally?

Comment (brikis98) [reply examples/ecs-fargate-service/.terraform.tfstate.lock.info]: Whoops, nope, good catch. Fixed: https://github.com/gruntwork-io/terragrunt-infrastructure-modules-example/pull/44/commits/3bb3ee2e8275f3fcf75d372cd633e5a5b10e59b5 and https://github.com/gruntwork-io/terragrunt-infrastructure-modules-example/pull/44/commits/36ed1c79c24db1390b2f535fb55a18db87ad5960.

**Review by Resonance1584 — APPROVED** [review]

**Comment by brikis98** [comment]
Thanks for the review! Merging now.

**Pull request merged**